### PR TITLE
feat: expose openapi spec under `_hub` secure routes

### PR DESCRIPTION
**Pull request by farnabaz**

### CODE REVIEW

File: src/module/runtime/server/api/_hub/openapi.get.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import openAPIJSON from '#internal/nitro/routes/openapi'
```
[the remark this answers is not in the document]
Comment (Atinux) [reply src/module/runtime/server/api/_hub/openapi.get.ts:1]: Can Nitro expose the openAPI then @pi0 ?

[the remark this answers is not in the document]
Comment (pi0) [reply src/module/runtime/server/api/_hub/openapi.get.ts:1]: It is in the roadmap for sure and i hope to have it for Nitro 2.10 out of the box which you can safely depend for production use.

**Comment by pi0** [comment]
If you want to introduce this feature before Nitro 2.10 via experimental flag:

- Make sure to check `nitro.options.experimental.openapi === true` to show the tab otherwise show instructions on how users can opt-in (please **never** implicitly enable an experimental flag for users!)

- Make sure to use dynamic import inside the handler and test if having an invalid virtual won't break the hub. Internal virtual can change all the time and this implementation shouldn't be broken if this 

- I don't see any authentication. Please make sure to authenticate with object syntax and explicit utility (instead of implicit middleware) to make sure the API is not unwillingly exposed.

**Comment by Atinux** [comment]
Thank you for the feedback 😊 

For the middleware we will refactor it to assign on each route in another PR, thanks for the reminder!
The middleware is https://github.com/nuxt-hub/core/blob/main/src/module/runtime/server/middleware/1.hub-auth.ts

**Review by Atinux — APPROVED** [review]

**Pull request merged**